jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -49,12 +49,34 @@ public void visit(MethodNode mth) {
 			for (int i = 0; i < list.size(); i++) {
 				InsnNode modInsn = simplifyInsn(mth, list.get(i));
 				if (modInsn != null) {
+					if (i != 0 && modInsn.contains(AFlag.ARITH_ONEARG)) {
+
+						InsnNode mergedNode = simplifyOneArgConsecutive(
+								list.get(i - 1), list.get(i), (ArithNode) modInsn);
+
+						if (mergedNode != null) {
+							list.remove(i - 1);
```

Comment:
You can't just remove insn, because it results can be used somewhere else.
Check that it used only by your insn like this `list.get(i - 1).getResult().getSVar().getUseCount() == 2`

Reply:
Done in method, although we do the check for `modInsn.contains(AFlag.ARITH_ONEARG)`, which won't happen unless we pass through https://github.com/skylot/jadx/blob/master/jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java#L341